k8ssandra-operator sub chart [K8SSAND-819]

burmanm opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

**What this PR does**:
Adds a new chart that deploys k8ssandra-operator. Can be enabled with k8ssandra-operator/enabled=true in the values.yaml

**Which issue(s) this PR fixes**:
Fixes #1046 , https://github.com/k8ssandra/k8ssandra-operator/issues/98

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

adejanovski commented on charts/k8ssandra-operator/values.yaml:
```diff
@@ -0,0 +1,73 @@
+  # -- Mark root filesystem as read only
+  readOnlyRootFilesystem: true
+
+  # -- Run cass-operator container as non-root user
```
Should it be `k8ssandra-operator` instead of `cass-operator`?

adejanovski reviewed (approved):
I was able to deploy K8ssandra with and without k8ssandra-operator by changing the `k8ssandra-operator/enabled` setting in the values file, as expected 👍 

just a nit about a comment in the k8ssandra-operator values file which mentions cass-operator.

burmanm commented:
Fixed the comment, disabled the k8ssandra-operator by default and added a small comment that people shouldn't use the k8ssandra-operator yet in production (this is just an experimental option at this point).
